### ISSUE TITLE
PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 failing on waterman cuda 9.2 builds

### ISSUE THREAD

**Issue by fryeguy52** [trilinos/Trilinos#3340]
CC: @trilinos/panzer , @mperego  (Trilinos Discretizations Product Lead), @bartlettroscoe 
 
## Next Action Status

Test was disabled in PR #3559 merged on 9/26/2017 and test is not showing up as failing in 'waterman' `cuda-9.2` builds on 9/27/2018 as shown [here](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-09-27&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-waterman-&field2=status&compare2=62&value2=passed).   PR #3562 merged on 10/8/2018 should have fixed this but the test is still disabled so we can't verify that.  PR #3642 merged on 10/16/2018 that re-enables this test in the `cuda-9.2` builds on 'waterman'.

Next: Watch for this test running in these builds on 10/17/2018 ...

## Description

As shown in [this query](https://testing-vm.sandia.gov/cdash/queryTests.php?project=Trilinos&date=2018-08-20&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercombine=and&filtercount=4&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-waterman-&field2=buildname&compare2=63&value2=cuda&field3=status&compare3=62&value3=passed&field4=status&compare4=62&value4=notrun) the test:

* PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4

is failing in the builds:

* Trilinos-atdm-waterman-cuda-9.2-opt
* Trilinos-atdm-waterman-cuda-9.2-debug

For more details, see close duplicate issue #3372.

## Steps to Reproduce

One should be able to reproduce this failure on the machine waterman as described in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

More specifically, the commands given for the system waterman are provided at:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md#waterman

The exact commands to reproduce this issue should be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh cuda-debug

$ cmake \
  -GNinja \
  -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
  -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_Panzer=ON \
  -DPanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4_DISABLE=OFF \
  $TRILINOS_DIR

$ make NP=20

$ bsub -x -Is -n 20 ctest -j20
```

**Comment by mperego** [comment]
@fryeguy52, Did the test ever pass on Waterman?

**Comment by mperego** [comment]
This might be related to Kokkos. @ndellingwood is looking into this.

**Comment by ndellingwood** [comment]
I'll test builds of kokkos, kokkos-kernels with the same modules etc. as a baseline.

**Comment by rppawlo** [comment]
So two tests are running out of memory in the cuda device. Are there other executables running at the same time? If so we may need to cut back on jobs overloading the cuda cards. If not then maybe we should just disable for this platform for now.

The two other convergence tests are timing out. It looks like this is due to kokkos support in muelu and xpetra not being enabled in the builds. See @jhux2 's comment in issue #3341 .

**Comment by bartlettroscoe** [comment]
@rppawlo are you referring to the Kokkos refactor options in #2674?

**Comment by rppawlo** [comment]
yes.

**Comment by ndellingwood** [comment]
FWIW Kokkos builds and tests pass with the master and develop VOTD branches. 

I tested with modules set up like in `/ascldap/users/ndellin/Trilinos/cmake/std/atdm/waterman/environment.sh`, i.e. 

```
module load devpack/20180517/openmpi/3.1.0/gcc/7.2.0/cuda/9.2.88
module swap openblas/0.2.20/gcc/7.2.0 netlib/3.8.0/gcc/7.2.0
```

Then generating the makefile as 
`../../generate_makefile.bash --with-cuda --with-openmp --arch="Power9,Volta70" --compiler="${HOME}/kokkos/bin/nvcc_wrapper" --prefix=`pwd`/install`

**Comment by bartlettroscoe** [comment]
FYI: @rppawlo, @bathmatt

**Comment by bartlettroscoe** [comment]
FY: @nmhamster says that this test looks like it is running out of memory on the GPU.

@rppawlo and @bathmatt, therefore, can we just disable this test for these builds on 'waterman'?  Otherwise, if we try to reduce the number of tests we run at the same time it may greatly increase the wallclock time for running all of the tests.  Note that as shown [here](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-09-24&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=buildname&compare2=63&value2=cuda&field3=testname&compare3=61&value3=PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4) this test is running and passing in the other GPU builds on 'white', 'ride', and 'hansen' (i.e 'shiller').

We can only fight so many battles.

**Comment by bartlettroscoe** [comment]
> @bartlettroscoe - so I think one option is to set these particular tests to use the maximum amount of cores/etc we setup for the testing. This would mean these are run serially but all the other tests could run in parallel? Maybe this will get these to pass and still enable some parallelism elsewhere. What say you?

@nmhamster, so CTest actually has a test property called `RUN_SERIAL` that should result in the test getting run by itself and there is direct support for that in [TRIBITS_ADD_ADVANCED_TEST()](https://tribits.org/doc/TribitsDevelopersGuide.html#tribits-add-advanced-test) used by this test.  But note that this test takes upwards of 10 minutes to pass in other builds as shown [here](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-09-24&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=buildname&compare2=63&value2=cuda&field3=testname&compare3=61&value3=PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4) so this may increase the wall-clock time by up to 10 minutes.  Currently, the cuda-9.2 builds on 'waterman' are taking about an hour to run.  This will also impact the time to run a CUDA CI build (see #2464).

@rppawlo, I would like to propose that we try adding the `RUN_SERIAL` test property only in the CUDA builds on 'waterman'.  To do that cleanly, I can add support for that in TriBITS to allow us to set this option in the ATDM `*.cmake` tweaks files.

**Comment by bartlettroscoe** [comment]
But first we need to verify that running `PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4` by itself on 'waterman' in these cuda-9.2 builds allows it to pass.  I will do that right now.

**Comment by bartlettroscoe** [comment]
@rppawlo and @nmhamster,

I tried running just this one test by itself on 'waterman' with:

```
$ bsub -x -Is -n 20 \
  ./checkin-test-atdm.sh cuda-9.2-opt-Power9-Volta70 cuda-9.2-debug-Power9-Volta70 \
  --enable-packages=Panzer \
  --ctest-options="-R PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4" \
  --local-do-all
```

and it returned:

**cuda-9.2-opt-Power9-Volta70:**
```
Test project /home/rabartl/Trilinos.base/BUILDS/WATERMAN/CHECKIN/cuda-9.2-opt-Power9-Volta70
    Start 132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4
1/1 Test #132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 ...***Failed  Required regular expression not found.Regex=[OVERALL FINAL RESULT: TEST PASSED .PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4.
] 27.97 sec

0% tests passed, 1 tests failed out of 1

Subproject Time Summary:
Panzer    = 111.90 sec*proc (1 test)

Total Test time (real) =  28.19 sec

The following tests FAILED:
	132 - PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 (Failed)
Errors while running CTest
```

**cuda-9.2-debug-Power9-Volta70:** 
```
Test project /home/rabartl/Trilinos.base/BUILDS/WATERMAN/CHECKIN/cuda-9.2-debug-Power9-Volta70
    Start 132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4
1/1 Test #132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 ...***Failed  Required regular expression not found.Regex=[OVERALL FINAL RESULT: TEST PASSED .PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4.
] 37.02 sec

0% tests passed, 1 tests failed out of 1

Subproject Time Summary:
Panzer    = 148.06 sec*proc (1 test)

Total Test time (real) =  37.32 sec

The following tests FAILED:
	132 - PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 (Failed)
Errors while running CTest
```

The errors start in:

```
================================================================================

TEST_2

Running: "/home/projects/ppc64le-pwr9-nvidia/openmpi/2.1.2/gcc/7.2.0/cuda/9.2.88/bin/mpiexec" "-np" "4" "-map-by" "socket:PE=4" "/home/rabartl/Trilinos.base/BUILDS/WATERMAN/CHECKIN/cuda-9.2-opt-Power9-Volta70/packages/panzer/adapters-stk/example/CurlLaplacianExample/PanzerAdaptersSTK_CurlLaplacianExample.exe" "--use-epetra" "--use-twod" "--cell=Quad" "--x-elements=16" "--y-elements=16" "--z-elements=4" "--basis-order=4"

  Writing output to file "/home/rabartl/Trilinos.base/BUILDS/WATERMAN/CHECKIN/cuda-9.2-opt-Power9-Volta70/packages/panzer/adapters-stk/example/CurlLaplacianExample/MPE-ConvTest-Quad-4-16"

--------------------------------------------------------------------------------

[waterman1:73334] mca_base_component_repository_open: unable to open mca_sec_munge: libmunge.so.2: cannot open shared object file: No such file or directory (ignored)
[waterman1:73340] mca_base_component_repository_open: unable to open mca_sec_munge: libmunge.so.2: cannot open shared object file: No such file or directory (ignored)
[waterman1:73341] mca_base_component_repository_open: unable to open mca_sec_munge: libmunge.so.2: cannot open shared object file: No such file or directory (ignored)
[waterman1:73342] mca_base_component_repository_open: unable to open mca_sec_munge: libmunge.so.2: cannot open shared object file: No such file or directory (ignored)
[waterman1:73343] mca_base_component_repository_open: unable to open mca_sec_munge: libmunge.so.2: cannot open shared object file: No such file or directory (ignored)
Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name waterman1 and rank 0!
Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name waterman1 and rank 1!
Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name waterman1 and rank 2!
Teuchos::GlobalMPISession::GlobalMPISession(): started processor with name waterman1 and rank 3!
:0: : block: [0,11,12], thread: [6,1,1] Assertion `Allocation failed.` failed.
:0: : block: [0,11,12], thread: [7,1,1] Assertion `Allocation failed.` failed.
:0: : block: [0,11,12], thread: [8,1,1] Assertion `Allocation failed.` failed.
:0: : block: [0,11,12], thread: [9,1,1] Assertion `Allocation failed.` failed.
:0: : block: [0,3,1], thread: [4,1,1] Assertion `Allocation failed.` failed.
...
```

Therefore, running this test by itself does not fix this problem.

Can temporarily disable this test for the CUDA builds on waterman?  Then some Panzer developer can debug this issue offline.  This issue will stay open with the "Disabled Tests" label in that case.

**Comment by rppawlo** [comment]
Sure.

**Comment by bartlettroscoe** [comment]
After talking with @bathmatt, I closed the later duplicate issue #3372.  However, that issue has a lot of good conversation that can be referenced there.

Also, @bathmatt says that more analysis of this failing test shows that it is just trying to allocate too much memory.

NOTE: This test is currently disabled so to enable it, you need to configure with:

```
  -D PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4_DISABLE=OFF
```

I updated the above "Steps to Reproduce" to handle this.  And once this test is fixed, it can be re-enabled by reverting commit f17f9bd27fde393e92064917525998b3a3b9f1ff on your topic branch and then of course create a PR and merge.

**Comment by rppawlo** [comment]
It looks like this is an issue with Intrepid2 when using sacado FAD types (@mperego @kyungjoo-kim ). The failing function is:
```
    Intrepid2::FunctionSpaceTools<PHX::Device::execution_space>::
      HCURLtransformVALUE(basis_vector.get_view(),
                          jac_inv.get_view(),
                          basis_ref_vector.get_view());
```
The error looks to be in apply_matvec_product() - lines 713 and 720 of the file Intrepid2_ArrayToolsDefTensor.hpp. These lines create temporary scalar variables (tmp) that don't know how to size the sacado FAD array. By converting all code to use BasisValues2\<double\> instead of BasisValues2\<FAD\<double\>\>, I could get all tests passing. We currently don't have any use cases where the coordinates are DOFs so it looks ok for panzer to switch all this code to use BasisValues2\<double\> instead of BasisValues2\<ScalarT\>.  Note that we only see test failures for higher order basis (4th order) where derivative array gets large. I'm guessing that this means that the fad type might have a default size that works for small basis dimensions (@etphipp )?

Some comments:
1. @bathmatt noted that the compile time of BasisValues2 object file with cuda was on the order of 45 minutes. I split out the instantiations to time separately. BasisVales2\<double\> takes less than a minute. BasisVales2\<Fad\<double\>\> takes 35-45 minutes. I've removed the FAD version from panzer to reduce cuda build times. All uses of BasisValues2 in panzer have been changed over to double. This not only will speed up compiletimes but also runtimes (when not using the fast sensitivity option).
1. Its highly likely this will fix #2751 for the opt build, although it is still timing out due to long runtimes. It no longer seg faults.
1. Still need to test against Charon and Drekar to make sure they are not directly using BasisValues2<FAD<...>> anywhere in their code. If so need to make corresponding changes.
1. Changed over the CurlLaplacian test to use tpetra instead of epetra. Runtimes differences are below:

Epetra runtimes for cuda waterman build:
```
bash-4.2$ ctest -R Curl
Test project /ascldap/users/rppawlo/Trilinos/BUILD_WATERMAN_DEBUG/packages/panzer
    Start 128: PanzerAdaptersSTK_CurlLaplacianExample
1/5 Test #128: PanzerAdaptersSTK_CurlLaplacianExample .........................   Passed   13.11 sec
    Start 129: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-1
2/5 Test #129: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-1 ...   Passed   21.97 sec
    Start 130: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-2
3/5 Test #130: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-2 ...   Passed   24.70 sec
    Start 131: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-3
4/5 Test #131: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-3 ...   Passed   37.18 sec
    Start 132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4
5/5 Test #132: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 ...   Passed  119.53 sec

100% tests passed, 0 tests failed out of 5

Label Time Summary:
Panzer    = 865.99 sec*proc (5 tests)

Total Test time (real) = 216.60 sec
bash-4.2$ 
```
Tpetra runtimes for cuda waterman build:
```
bash-4.2$ ctest -R Curl
Test project /ascldap/users/rppawlo/Trilinos/BUILD_WATERMAN_DEBUG/packages/panzer/adapters-stk/example/CurlLaplacianExample
    Start 1: PanzerAdaptersSTK_CurlLaplacianExample
1/5 Test #1: PanzerAdaptersSTK_CurlLaplacianExample .........................   Passed   12.60 sec
    Start 2: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-1
2/5 Test #2: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-1 ...   Passed   23.62 sec
    Start 3: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-2
3/5 Test #3: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-2 ...   Passed   29.69 sec
    Start 4: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-3
4/5 Test #4: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-3 ...   Passed   78.21 sec
    Start 5: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4
5/5 Test #5: PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4 ...   Passed  188.70 sec

100% tests passed, 0 tests failed out of 5

Label Time Summary:
Panzer    = 1331.27 sec*proc (5 tests)

Total Test time (real) = 332.85 sec
bash-4.2$ 

```

**Comment by etphipp** [comment]
I don't think the issue is setting the Fad dimension.  These are just scalars, which get their size from the right-hand-side of the assignment.  You only need to worry about the Fad dimension when you have a view of Fads.

Rather, what I think is happening is you are running out of heap space on the GPU, since I am guessing you are using a dynamically-sized Fad type (i.e., DFad).  The GPU by default only provides a tiny amount of heap space for allocating memory dynamically in kernels.  When you have a larger Fad dimension, you are probably exhausting this memory.  You can fix this by not using a dynamically sized Fad type (e.g., SLFad) or increase the amount of heap space via `cudaDeviceSetLimit(cudaLimitMallocHeapSize, size_t size)`.  See here for more info:

https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#dynamic-global-memory-allocation-and-operations

**Comment by etphipp** [comment]
Btw, any Panzer kernel that has temporary scalars and uses DFad will have this problem.

**Comment by rppawlo** [comment]
Thanks @etphipp ! That explains it! So far, this is the only instance in the entire panzer test suite where it occurs (very surprising). It's interesting that this happens on waterman with volta, but not on shiller with K80. I would have expected the default memory allowance would grow on newer cuda cards. Wonder how the default is set...

**Comment by mperego** [comment]
Thanks @rppawlo and @etphipp for figuring this out!

**Comment by etphipp** [comment]
I'm not sure how the default is set, but one thing that could happen is there might be more threads running concurrently on volta than an k80.  The heap space is fixed, independent of the number of threads, effectively giving you a smaller amount per thread on volta.

**Comment by bartlettroscoe** [comment]
FYI: PR #3559 will disable this test in the 'waterman' `cuda-9.2` builds for now once it gets merged.  Then once this is fixed, we can revert the disable commit and get this test running again (perhaps even on other builds as well where it was disabled).

**Comment by rppawlo** [comment]
The Curl tests have been reenabled and have been passing for the last few days. Closing.

**Comment by bartlettroscoe** [comment]
@rppawlo, can we leave up just until we get confirmation on CDash tomorrow?

**Comment by bartlettroscoe** [comment]
@rppawlo, what Trilinos PR fixed this?

**Comment by rppawlo** [comment]
#3562 - @jmgate merged this last week while I was on vacation. A different failure was logged in another ticket #3579 by @fryeguy52 last week as well.

**Comment by bartlettroscoe** [comment]
Note that as of today, this test is still timing out in the CUDA debug and opt builds on 'hansen' as shown [here](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-10-15&filtercount=3&showfilters=1&filtercombine=and&field1=groupname&compare1=61&value1=ATDM&field2=testname&compare2=65&value2=Panzer&field3=status&compare3=62&value3=passed).  Is this test not failing in the EMPIRE Trilinos builds on 'shiller' as well?

**Comment by rppawlo** [comment]
So the title of this ticket is to fix the failing waterman build. It was failing due to running out of memory on device and crashing. We fixed that and verified that it passed on waterman. That's why I closed this ticket. Once the memory issue was fixed, we are now seeing timeouts on hansen and white (mainly because I reenabled this test everywhere hoping that this fix cleaned up all machines). A new ticket was opened to address these timeouts - #3579. If we are leaving this open to address the timeouts on hansen and white, then I guess we should close #3579 as a duplicate?

**Comment by bartlettroscoe** [comment]
@rppawlo, this test got disabled in the waterman CUDA builds in PR  #3559  merged on 9/26/2017 (see the above "Next Action Status" field).  We would need to re-enable this test in this builds by removing:

```
# This test runs out of CUDA memory all on its own (#3340)
ATDM_SET_ENABLE(PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4_DISABLE ON)
```

from the file `Trilinos/cmake/std/atdm/waterman/tweaks/CUDA_COMMON_TWEAKS.cmake` that was added in commit aab6928c2c62d3be1add453ffaeff4f2f50a6f6b.  Can someone create a PR that does that?

NOTE: We try to keep the "Next Action Status" field up to date so that the current status of an issue is clear.

**Comment by bartlettroscoe** [comment]
@rppawlo, sorry, you are right about #3579 covering the timeouts.  Once this test is re-enabled in the `cuda-9.2` builds on 'waterman', then we can extend the scope of #3579 to cover timeouts on 'waterman' (in case that happens) and then we can close this issue (which is about the CUDA memory failure).

Sorry about the confusion.  There are a lot of outstanding Trilinos GitHub issues for ATDM Trilinos failures right now and it is hard to keep everything straight with things constantly changing.  Hopefully by mid next week, we will have #2933 far enough along that @fryeguy52 and I can keep better track of all of these builds and failures.

**Comment by rppawlo** [comment]
Sorry - I thought I reverted all disables of this test but somehow missed the waterman one. The Curl example is running for opt builds in 140 seconds on waterman for 4th order in my testing so it should now be passing. Guess the older hardware is impacting things on the other machines? I will put up a PR shortly to reenable.

**Comment by bartlettroscoe** [comment]
PR #3642 merged on 10/16/2018 that re-enables this test in the `cuda-9.2` builds on 'waterman'.  Next: Watch for this test running in these builds on 10/17/2018.  Putting this issue "in review" ...

**Comment by bartlettroscoe** [comment]
> The Curl example is running for opt builds in 140 seconds on waterman for 4th order in my testing so it should now be passing. Guess the older hardware is impacting things on the other machines? I will put up a PR shortly to reenable.

@rppawlo, we see some significant slowdown for some tests with CUDA builds on some machines with `ctest -j8` because all of the MPI processes are running on the same GPU device.  (But again, the overall wall-clock time goes down by a good bit with `ctest -j8` vs. `ctest -j1` so it is worth running tests in prallel overall.)  Let's see how this goes tomorrow.

Thanks for posting that PR!

**Comment by bartlettroscoe** [comment]
As shown in [this query](https://testing.sandia.gov/cdash-dev-view/queryTests.php?project=Trilinos&date=2018-10-17&filtercount=3&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-waterman-cuda-9.2-&field2=testname&compare2=61&value2=PanzerAdaptersSTK_CurlLaplacianExample-ConvTest-Quad-Order-4&field3=buildstarttime&compare3=83&value3=6%20weeks%20ago), this test before it was disabled last ran on 9/26/2018 and then when it was re-enabled yesterday we see it ran today and passed in both of the `cuda-9.2-debug` and `cuda-9.2-opt` builds.

Therefore, this issue has been resolved and we can close this issue.

Thanks @rppawlo for fix this!
